build_system/seqan3-config.cmake

```
@@ -250,30 +250,33 @@ set (CXXSTD_TEST_SOURCE
     "static_assert (__cpp_concepts >= 201507);
     int main() {}")
 
-check_cxx_source_compiles ("${CXXSTD_TEST_SOURCE}" CONCEPTS_BUILTIN)
+set (SEQAN3_FEATURE_CONCEPT_FLAG_BUILTIN "")
+set (SEQAN3_FEATURE_CONCEPT_FLAG_STD20 "-std=c++20")
+set (SEQAN3_FEATURE_CONCEPT_FLAG_STD2a "-std=c++2a")
+set (SEQAN3_FEATURE_CONCEPT_FLAG_GCC_TS "-fconcepts")
 
-if (CONCEPTS_BUILTIN)
-    seqan3_config_print ("C++ Concepts support:       builtin")
-else ()
-    set (CONCEPTS_FLAG "")
-
-    foreach (_FLAG -std=c++20 -std=c++2a -fconcepts)
-        set (CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS_SAVE} ${_FLAG}")
+set (CONCEPTS_FLAG "")
```

```suggestion
set (SEQAN3_CONCEPTS_FLAG "")
```

can we prefix this, as it is the final result?

The intermediate variables (as `CONCEPTS_FLAG${_FLAG}`) don't necessarily need to be prefixed.